packages/nc-gui-v2/components/smartsheet/sidebar/menu/ApiSnippet.vue

```diff
@@ -0,0 +1,183 @@
+<script setup lang="ts">
+import HTTPSnippet from 'httpsnippet'
+import { useClipboard } from '@vueuse/core'
+import { useToast } from 'vue-toastification'
+import { ActiveViewInj, MetaInj } from '~/context'
+
+const props = defineProps<Props>()
+
+const emits = defineEmits(['update:modelValue'])
+
+const toast = useToast()
+
+interface Props {
+  modelValue: boolean
+}
+
+const { project } = $(useProject())
+const { appInfo } = $(useGlobal())
+const meta = $(inject(MetaInj))
+const view = $(inject(ActiveViewInj))
+const { xWhere } = useSmartsheetStoreOrThrow()
+const { queryParams } = $(useViewData(meta, view as any, xWhere))
+const { copy } = useClipboard()
+const { $state } = useNuxtApp()
```

Comment:
This is a double call technically.
`useGlobal` will return `$state` if it's already available, so you can just get `token` etc. from `useGlobal`

```ts
const { token, appInfo } = $(useGlobal())
```